File: sdks/remote_plugin/python/app/schema/payload.py
from __future__ import annotations

from dataclasses import dataclass, field
from typing import Any

from app.schema.base import Properties, decode_json_bytes, properties_from_model

PAYLOAD_ASR_RESULT = "asr_result"
PAYLOAD_CAPTION = "caption"
PAYLOAD_LLM_CHUNK = "llm_chunk"


@dataclass(slots=True)
class Payload(Properties):
    name: str
    properties: dict[str, Any] = field(default_factory=dict)

    @classmethod
    def from_json_bytes(cls, name: str, data: bytes) -> Payload:
        return cls(name=name, properties=decode_json_bytes(data))

    @classmethod
    def from_model(cls, name: str, value: Any) -> Payload:
        return cls(name=name, properties=properties_from_model(value))

File: sdks/remote_plugin/nodejs/src/schema/payload.ts
import { Properties, decodeJsonBytes } from './base.js'

// Well-known payload names
export const PAYLOAD_ASR_RESULT = 'asr_result'
export const PAYLOAD_CAPTION = 'caption'
export const PAYLOAD_LLM_CHUNK = 'llm_chunk'

export class Payload extends Properties {
  /** Create a Payload from a name and JSON-encoded bytes. */
  static fromJsonBytes(name: string, data: Buffer | Uint8Array): Payload {
    return new Payload(name, decodeJsonBytes(data))
  }

  /** Create a Payload from a name and a plain object. */
  static fromObject(name: string, obj: Record<string, unknown>): Payload {
    return new Payload(name, obj)
  }
}
